examples/rbfe_edge_list.py

```diff
@@ -108,25 +107,44 @@ def read_from_args():
 
     for fut, meta in zip(futures, metadata):
         solvent_res, solvent_top, complex_res, complex_top = fut.result()
-        solvent_ddg = np.sum(solvent_res.all_dGs)
-        solvent_ddg_err = np.linalg.norm(solvent_res.all_errs)
-        complex_ddg = np.sum(complex_res.all_dGs)
-        complex_ddg_err = np.linalg.norm(complex_res.all_errs)
-
-        mol_a, mol_b, _, _, exp_ddg, fep_ddg, fep_ddg_err, ccc_ddg, ccc_ddg_err = meta
-        mol_a_name = get_mol_name(mol_a)
-        mol_b_name = get_mol_name(mol_b)
 
         with open(f"rbfe_result_{mol_a_name}_{mol_b_name}.pkl", "wb") as fh:
             pkl_obj = (meta, solvent_res, solvent_top, complex_res, complex_top)
             pickle.dump(pkl_obj, fh)
 
-        tm_ddg = complex_ddg - solvent_ddg
-        tm_err = np.linalg.norm([complex_ddg_err, solvent_ddg_err])
+        if solvent_res.exception:
+            exception = "".join(
+                traceback.format_exception(
+                    type(solvent_res.exception), solvent_res.exception, solvent_res.exception.__traceback__
+                )
+            )
```

Comment:
Idea to make this more "pythonic": instead of catching errors in `estimate_relative_free_energy` and passing them as values in the result object, could it work to catch and handle exceptions at the top layer (here)? E.g. thinking of something like this:
```suggestion
        try:
            solvent_res, solvent_top, complex_res, complex_top = fut.result()
        except FreeEnergyEstimationError: # custom exception class optional
            print(f"failed: {mol_a_name} -> {mol_b_name} solvent leg")
            traceback.print_exc()
```

Reply:
When I wrote this i wasn't sure how exception propagation is handled through multiprocessing - so I wanted to store the exception objection explicitly. I'll play around and see

Reply:
I'm not sure how to do this properly still - let's offline this 2mrw